### DISCUSSION TITLE
More info about key derivation function

### DISCUSSION THREAD

**Discussion by Bugaddr** [Kunzisoft/KeePassDX#1247] [Q&A] [answered]
Hello all iam currently using keepassdx as my password manager, i want  know which will be best key derivation function for better security of my database.
There are three options
AES, ARGON2D, ARGON2ID

I am currently on 
Rijndael(aes) encryption algorithm
25 key derivation rounds
I am using keyfile

Sorry, but i have very less information regarding how encryption works.

**Comment by J-Jamet** [comment, marked as answer]
AES : https://en.wikipedia.org/wiki/Advanced_Encryption_Standard
Argon2 : https://en.wikipedia.org/wiki/Argon2

If you have no particular constraint and if you use another application compatible with the kdbx v4.0 format. You can use Argon2id as KDF.
